packages/medusa/src/loaders/plugins.ts

```diff
@@ -345,12 +391,20 @@ export async function registerServices(pluginDetails: PluginDetails, container:
           ).singleton(),
           [`tp_${loaded.identifier}`]: aliasTo(name),
         })
-      } else {
+      } else if (
+        loaded.prototype instanceof BaseService ||
+        loaded.prototype instanceof TransactionBaseService
+      ) {
         container.register({
           [name]: asFunction(
             (cradle) => new loaded(cradle, pluginDetails.options)
           ),
         })
+      } else {
+        const logger = container.resolve<Logger>("logger")
+        const message = `Services must inherit from BaseService, please check ${fn}`
```

Comment:
question: I think that since we are loading different type of object here, this message should be rewritten wdyt?